oxide-auth/src/primitives/generator.rs

```diff
@@ -232,21 +241,30 @@ impl TagGrant for RandomGenerator {
     }
 }
 
-impl<'a> TagGrant for &'a RandomGenerator {
-    fn tag(&mut self, _: u64, _: &Grant) -> Result<String, ()> {
-        Ok(self.generate())
-    }
-}
+// impl<'a> TagGrant for &'a RandomGenerator {
+//     fn tag(&mut self, _: u64, _: &Grant) -> Result<String, ()> {
+//         // Ok(self.generate())
+//         Err(())
+//     }
+// }
 
 impl TagGrant for Rc<RandomGenerator> {
     fn tag(&mut self, _: u64, _: &Grant) -> Result<String, ()> {
-        Ok(self.generate())
+        if let Some(rng) = Rc::get_mut(self) {
+            Ok(rng.generate())
+        } else {
+            Err(())
```

Comment:
Why remove on impl but let the other error? Using `Rc::get_mut` is just brittle.

Reply:
You were right with the `&self mut` on `generate`. I reverted and fixed this.